lib/src/barback/dartdevc/module_computer.dart

```diff
@@ -353,6 +354,26 @@ class _ModuleComputer {
     return modulesById.values.toList();
   }
 
+  /// Renames shared [Module]s to something unique and short to avoid issues
+  /// with file names that are too long.
+  List<Module> _renameSharedModules(List<Module> modules) {
+    if (modules.isEmpty) return modules;
+    var next = 0;
+    // All modules and assets in those modules share a top level dir, we just
+    // grab it from the first one.
```

Comment:
It would be good to move this up to the main doc comment and clarify that it's a requirement of the argument.